astropy/units/format/base.py

```diff
@@ -61,7 +62,21 @@ def format_exponential_notation(
         str
             The value in exponential notation in a this class's format.
         """
-        return format(val, format_spec)
+        x = format(val, format_spec).split("e")
+        if len(x) != 2:
+            return cls._format_mantissa(x[0])
+        ex = x[1].lstrip("0+")
+        if not ex:
+            return cls._format_mantissa(x[0])
+        if ex.startswith("-"):
+            ex = "-" + ex[1:].lstrip("0")
+        ex = f"10{cls._format_superscript(ex)}"
+        m = cls._format_mantissa(x[0] if x[0] != "1." + "0" * (len(x[0]) - 2) else "")
+        return f"{m}{cls._times}{ex}" if m else ex
```

Comment:
Some comments would be great in this section. E.g. when does ``ex = "-" + ex[1:].lstrip("0")`` happen?

Reply:
There were no comments in the previous code and the new code is even simpler. But we could consider replacing
```patch
- m = cls._format_mantissa(x[0] if x[0] != "1." + "0" * (len(x[0]) - 2) else "")
+ m = cls._format_mantissa(x[0] if x[0].rstrip("0") != "1." else "")
```

Reply:
That's a good replacement.
But I still thinking adding comments would be good. This is a lot of string manipulation.

Reply:
Like the suggested replacement; comments are OK, though to me the various steps are fairly self explanatory; e.g., why the `0` needs to be removed:

```
In [1]: f"{-0.1:.3e}"
Out[1]: '-1.000e-01'
```

Reply:
I still don't think comments are really needed here, but since they were requested I added a couple.